File: 프로그래머스/unrated/250135. ［PCCP 기출문제］ 3번 ／ 아날로그 시계/［PCCP 기출문제］ 3번 ／ 아날로그 시계.py
def solution(h1, m1, s1, h2, m2, s2):
    def get_count(h, m, s):
        m_count = h * 59 + m
        h_count = h * 60 + m

        result = -1

        cur_m_degree = m * 6
        cur_h_degree = 30 * (h % 12) + 0.5 * m

        if cur_m_degree <= 5.9 * s:
            m_count += 1
        
        if cur_h_degree <= (6 - 1 / 120) * s:
            h_count += 1
        
        if h >= 12:
            h_count -= 1
            result -= 1

        return result + m_count + h_count

    answer = get_count(h2, m2, s2) - get_count(h1, m1, s1)
    return answer + (s1 == 0 and m1 == 0)

File: 프로그래머스/2/250135. ［PCCP 기출문제］ 3번 ／ 아날로그 시계/［PCCP 기출문제］ 3번 ／ 아날로그 시계.java
public class Solution {
    
    public int solution(int h1, int m1, int s1, int h2, int m2, int s2) {
        return getCount(h2, m2, s2) - getCount(h1, m1, s1) + (s1 == 0 && m1 == 0 && (h1 % 12) == 0 ? 1 : (s1 == 0 && m1 == 0 ? 1 : 0));
    }

    private int getCount(int h, int m, int s) {
        int mCount = h * 59 + m;
        int hCount = h * 60 + m;
        int result = -1;

        double curMDegree = m * 6.0;
        double curHDegree = 30.0 * (h % 12) + 0.5 * m + (1.0 / 120.0) * s;

        if (curMDegree <= 5.9 * s) {
            mCount++;
        }

        if (curHDegree <= (6.0 - 1.0 / 120.0) * s) {
            hCount++;
        }

        if (h >= 12) {
            hCount--;
            result--;
        }

        return result + mCount + hCount;
    }
}